test/topicFilters.spec.ts

```diff
@@ -431,3 +441,190 @@ describe("Send/Receive messages using boolean filters of subscription", function
     await testPeekMsgsLength(subscriptionClient, 0);
   });
 });
+
+describe("Send/Receive messages using sql filters of subscription", function(): void {
+  beforeEach(async () => {
+    await beforeEachTest();
+  });
+
+  afterEach(async () => {
+    await receiveOrders(defaultSubscriptionClient);
+    await testPeekMsgsLength(defaultSubscriptionClient, 0);
+
+    await afterEachTest();
+  });
+
+  it("SQL rule filter on the message properties", async function(): Promise<void> {
+    await subscriptionClient.addRule("SQLMsgPropertyRule", "sys.label = 'red'");
+
+    const rules = await subscriptionClient.getRules();
+    should.equal(rules.length, 1, "rule");
+    should.equal(rules[0].name, "SQLMsgPropertyRule");
+
+    await sendOrders();
+    const receivedMsgs = await receiveOrders(subscriptionClient);
+
+    should.equal(Array.isArray(receivedMsgs), true);
+    should.equal(receivedMsgs.length, 4);
+
+    await testPeekMsgsLength(subscriptionClient, 0);
+  });
+
+  it("SQL rule filter on the custom properties", async function(): Promise<void> {
+    await subscriptionClient.addRule("SQLCustomRule", "color='red'");
+
+    const rules = await subscriptionClient.getRules();
+    should.equal(rules.length, 1, "rule");
+    should.equal(rules[0].name, "SQLCustomRule");
```

Comment:
Maybe we can add a function that does the adding rule and doing the above 2 checks? This part is repeated in a few test cases